app/models/blog.rb

```diff
@@ -6,16 +6,26 @@ class Blog < ApplicationRecord
   has_many :liking_users, class_name: 'User', source: :user, through: :likings
 
   validates :title, :content, presence: true
+  validate :eyecatch_requires_premium
 
   scope :published, -> { where('secret = FALSE') }
 
   scope :search, lambda { |term|
-    where("title LIKE '%#{term}%' OR content LIKE '%#{term}%'")
+    term &&= ActiveRecord::Base.sanitize_sql_like(term)
```

Comment:
`&&=`はtermに値があればサニタイズする、という意味でしょうか？
ちょっと意図がわかりづらいので、

```suggestion
    term = ActiveRecord::Base.sanitize_sql_like(term.to_s)
```

のようにした方がまだ良いかもしれません。

ただし、変数（引数）への再代入はなるべく避けたいですね

https://bootcamp.fjord.jp/pages/avoid-re-assign